components/PlantList.tsx

```diff
@@ -1,24 +1,172 @@
 import React, { useEffect, useState } from 'react';
-import { getPlantSeasonality } from '@/api/supabase/queries/plantSeasonality';
+import { getAllPlants } from '@/api/supabase/queries/plants';
 import { Plant } from '@/types/schema';
 
-export const PlantList = () => {
+interface PlantListProps {
+  harvestSeason: string;
+  plantingType: string;
+  growingSeason: string;
+}
+
+export const PlantList = ({
+  harvestSeason,
+  plantingType,
+  growingSeason,
+}: PlantListProps) => {
   const [plants, setPlants] = useState<Plant[]>([]);
+  const growingSeasonToIndex = new Map<string, number[]>([
+    ['Spring', [2, 3, 4]],
+    ['Summer', [5, 6, 7]],
+    ['Fall', [8, 9, 10]],
+    ['Winter', [11, 0, 1]],
+  ]);
+
+  const monthToIndex = new Map<string, number>([
+    ['JANUARY', 0],
+    ['FEBRUARY', 1],
+    ['MARCH', 2],
+    ['APRIL', 3],
+    ['MAY', 4],
+    ['JUNE', 5],
+    ['JULY', 6],
+    ['AUGUST', 7],
+    ['SEPTEMBER', 8],
+    ['OCTOBER', 9],
+    ['NOVEMBER', 10],
+    ['DECEMBER', 11],
+  ]);
 
   useEffect(() => {
     const fetchPlantSeasonality = async () => {
-      const plantList = await getPlantSeasonality('Tennessee');
-      setPlants(plantList);
+      // gets plants in Tennessee by default
+      const plantList = await getAllPlants();
+      const us_state = 'TENNESSEE';
+      const filteredPlantList = plantList.filter(
+        plant => plant.us_state === us_state,
+      );
+      setPlants(filteredPlantList);
     };
 
     fetchPlantSeasonality();
   }, []);
 
+  // Check if growingSeason matches the plant's growing season
+  const checkGrowingSeason = (plant: Plant) => {
+    // Automatically returns true if selected growing season is ''
+    if (!growingSeason) {
+      return true;
+    }
+
+    // list of valid indexes for the growing season
+    // indexes are the months of the year
+    const validIndexes = growingSeasonToIndex.get(growingSeason);
+
+    const isInRange = (start: number, end: number, validIndexes: number[]) => {
+      // Checks if the start and end months are within the valid range
+      if (start <= end) {
+        return validIndexes.some(index => index >= start && index <= end);
+      } else {
+        // Handle wrap-around case (e.g. NOVEMBER to FEBRUARY)
+        return validIndexes.some(index => index >= start || index <= end);
+      }
+    };
+
+    // Handle late/early month logic -- Set late/early month to just the month
+    let indoorsStart = plant.indoors_start;
+    let indoorsEnd = plant.indoors_end;
+    let outdoorsStart = plant.outdoors_start;
+    let outdoorsEnd = plant.outdoors_end;
+
+    /// If field is not null and starts with 'LATE' or 'EARLY,
+    // get substring after 'LATE_ or 'EARLY_'
+    if (
+      indoorsStart &&
+      (indoorsStart.startsWith('LATE') || indoorsStart.startsWith('EARLY'))
+    ) {
+      indoorsStart = indoorsStart.substring(5);
+    }
```

Comment:
maybe create a util func that gets the month value, to repeat code repetition with indoorsStart, indoorsEnd etc.? 
Also, for 'EARLY', you'd probably need to do `indoorsStart.substring(6)` instead `indoorsStart.substring(5)`  so let's make separate checks for that!